apps/data-mapper/src/services/create.ts

```diff
@@ -280,3 +305,77 @@ export const createInformedConsent = async (
 		return failure('SYSTEM_ERROR', 'An unexpected error occurred.');
 	}
 };
+
+/**
+ * Creates a Consent for Research Participation Participant Response entry in the Consent DAS
+ *
+ * @async
+ * @param {ConsentResearchParticipationRequest} consentResearchParticipationRequest - Consent for Research Participation Participant Response request data
+ * @param {NanoId} participantId - Participant ID
+ * @returns {ConsentResearchParticipationResponse} ConsentResearchParticipationResponse object for consumption by the Consent API
+ */
+export const createConsentResearchParticipation = async (
+	consentResearchParticipationRequest: ConsentResearchParticipationRequest,
+	participantId: NanoId,
+): Promise<
+	Result<ConsentResearchParticipationResponse, CreateParticipantResponseFailureStatus>
+> => {
+	try {
+		// RESEARCH_PARTICIPATION__CONTACT_INFORMATION
+		const contactInformationResponseRequest = {
+			participantId,
+			consentQuestionId: RESEARCH_PARTICIPATION__CONTACT_INFORMATION,
+			response: consentResearchParticipationRequest[RESEARCH_PARTICIPATION__CONTACT_INFORMATION],
+		};
+		const contactInformationResponse = await createParticipantResponse(
+			contactInformationResponseRequest,
+		);
+		if (contactInformationResponse.status !== 'SUCCESS') {
+			return contactInformationResponse;
+		}
+		const parsedContactInformationResponse = ParticipantResponse.safeParse(
+			contactInformationResponse.data,
+		);
+		if (!parsedContactInformationResponse.success) {
+			logger.error(
+				'Received invalid data in create Consent for Research Participation - Contact Information response',
+				parsedContactInformationResponse.error.issues,
+			);
+			return failure('SYSTEM_ERROR', parsedContactInformationResponse.error.message);
+		}
+
+		// RESEARCH_PARTICIPATION__FUTURE_RESEARCH
+		const futureResearchResponseRequest = {
+			participantId,
+			consentQuestionId: RESEARCH_PARTICIPATION__FUTURE_RESEARCH,
+			response: consentResearchParticipationRequest[RESEARCH_PARTICIPATION__FUTURE_RESEARCH],
+		};
+		const futureResearchResponse = await createParticipantResponse(futureResearchResponseRequest);
+		if (futureResearchResponse.status !== 'SUCCESS') {
+			return futureResearchResponse;
+		}
+		const parsedFutureResearchResponse = ParticipantResponse.safeParse(futureResearchResponse.data);
+		if (!parsedFutureResearchResponse.success) {
+			logger.error(
+				'Received invalid data in create Consent for Research Participation - Future Research response',
+				parsedFutureResearchResponse.error.issues,
+			);
+			return failure('SYSTEM_ERROR', parsedFutureResearchResponse.error.message);
+		}
+
+		const consentResearchParticipation = {
+			[parsedContactInformationResponse.data.consentQuestionId]:
+				parsedContactInformationResponse.data.response,
+			[parsedFutureResearchResponse.data.consentQuestionId]:
+				parsedFutureResearchResponse.data.response,
+		};
+
+		return success(consentResearchParticipation);
```

Comment:
As mentioned in my comment on the response type, here's an example of validating the response type before returning from the create function. We don't **have** to do this, but we can check with Ann in office tomorrow and see if we should.

```suggestion
		const parsedConsentResearchParticipation = ConsentResearchParticipationResponse.safeParse(
			consentResearchParticipation,
		);
		if (!parsedConsentResearchParticipation.success) {
			logger.error(
				'Received invalid data in create Consent for Research Participation response',
				parsedConsentResearchParticipation.error.issues,
			);
			return failure('SYSTEM_ERROR', parsedConsentResearchParticipation.error.message);
		}

		return success(parsedConsentResearchParticipation.data);
```

Reply:
Discussed with @anncatton in office, we can proceed with these suggestions since we're combining separate response types into a single response, just to be safe that we're returning what we should 🙂